clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -437,6 +437,10 @@ static bool addExceptionArgs(const ArgList &Args, types::ID InputType,
     return false;
   }
 
+  if (Triple.isSPIR())
```

Comment:
```suggestion
  if (Triple.isSPIR() || Triple.isNVPTX() || Triple.isAMDGCN())
```
(we might even need a helper function like `isSYCLDeviceTarget()` for that, since similar conditions could come up elsewhere)